### ISSUE TITLE
'Go To Implementation' does not work with multi-targeting

### ISSUE THREAD

**Issue by sharwell** [dotnet/roslyn#43093]
**Version Used**: 16.6 Preview 2

**Steps to Reproduce**:

1. Create a project **BaseProject** with the following targets:

    ```xml
    <PropertyGroup>
      <TargetFrameworks>netcoreapp3.1;netstandard2.0</TargetFrameworks>
    </PropertyGroup>
    ```

1. Add the following interface to **BaseProject**

    ```csharp
    public interface IInterface
    {
    }
    ```

1. Create a project **ImplProject** with the following targets:

    ```xml
    <PropertyGroup>
      <TargetFramework>net472</TargetFramework>
    </PropertyGroup>
    ```

1. Add a reference from **ImplProject**&rarr;**BaseProject**

1. Add the following class to **ImplProject**

    ```csharp
    public class Impl : IInterface
    {
    }
    ```

1. Open **IInterface.cs** in **BaseProject**, and invoke **Go To Implementation** on `IInterface`

**Expected Behavior**:

The IDE navigates to `Impl` in **ImplProject**.

**Actual Behavior**:

An error is reported:

> The symbol has no implementations.

**Comment by sharwell** [comment]
@jinujoseph we need this for 16.6. The changes to multitarget **netcoreapp3.1** for improved behavior of nullable reference types have resulted in a serious impairment for Roslyn developers.

**Comment by CyrusNajmabadi** [comment]
i'm working on a general set of improvements right now that i think will fix thsi.
